dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,26 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA (val states: Set[State], val transitions: Set[Transition], val startState: State, val acceptingStates: Set[State]):
+    def accepts(s: String) =
+        var currentState = startState
+        var validTransition = transitions.head
+        var canTranslate = false
+        for (c <- s)
+            canTranslate = false
+            for (transition <- transitions)
+                if (currentState == transition.from && c == transition.symbol) {
+                    validTransition = transition
+                    canTranslate = true
+                }
+            if (canTranslate) {
```

Comment:
It feels like that if statement could be folded into the if statement of line 14, but then you'll have to trust the user to include exactly one transition per state/symbol pair